parlai/agents/transformer/modules.py

```diff
@@ -99,9 +108,26 @@ def __init__(self, opt, dictionary):
         if n_positions < 0:
             raise ValueError('n_positions must be positive')
 
+        if opt.get('gelu_activation'):
+            self.gelu_activation = opt['gelu_activation']
+        else:
+            self.gelu_activation = False
+
+        if opt.get('embedding_normalization'):
+            self.embedding_normalization = opt['embedding_normalization']
+        else:
+            self.embedding_normalization = False
+
+        if opt.get('reduction_type'):
```

Comment:
You can simplify these with `self.reduction_type = opt.get('reduction_type', False)`

Reply:
We could check here if --reduction is False and set reduction-type to None if that's the case?

Reply:
seems like we can remove the boolean and just have it be a fixed set. `{'none', 'mean', ...}`